docs/how_to_publish_to_cmr_with_earthdata_varinfo.ipynb

```diff
@@ -0,0 +1,314 @@
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```\n",
+    "pip install earthdata-varinfo\n",
+    "```\n",
+    "\n",
+    "If this doesn't work, alternatively you can clone the git repository, and install the package in editable mode:\n",
+    "\n",
+    "```\n",
+    "git clone https://github.com/nasa/earthdata-varinfo\n",
+    "cd earthdata-varinfo\n",
+    "pip install -e .\n",
+    "```"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Other notebook requirements:\n",
+    "\n",
+    "Required packages for `earthdata-varinfo` should be installed as dependencies of PyPI. For local development, without a standard pip installation, third party requirements can be installed from the following files:\n",
+    "\n",
+    "```\n",
+    "pip install -r requirements.txt -r dev-requirements.txt\n",
+    "pip install notebook\n",
+    "```"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Current output format\n",
+    "Currently `earthdata-varinfo` produces Python classes that can be used within Python scripts, Jupyter notebooks or a Python Read-Eval-Print Loop (REPL).\n",
+    "\n",
+    "`earthdata-varinfo` is also able to create UMM-Var compliant JSON records and publish these records to NASA's Common Metadata Repository (CMR)."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Example usage:\n",
+    "* [M2I1NXASM](https://cmr.uat.earthdata.nasa.gov/search/collections.umm_json?concept-id=C1256535511-EEDTEST)\n",
+    "* [GLDAS_NOAH10_3H](https://cmr.uat.earthdata.nasa.gov/search/collections.umm_json?concept-id=C1256543837-EEDTEST)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import CMR environment\n",
+    "from cmr import CMR_UAT\n",
+    "\n",
+    "# Import varinfo cmr_search functions\n",
+    "from varinfo.cmr_search import (get_granules, get_granule_link, \n",
+    "                                download_granule)\n",
+    "\n",
+    "# Import VarInfoFromNetCDF4 class and umm_var functions\n",
+    "# for reading and netCDF4 files and creating UMM-Var entries \n",
+    "from varinfo import VarInfoFromNetCDF4\n",
+    "\n",
+    "# Import functions for creating UMM-Var entry and\n",
+    "# publishing a UMM-Var entry\n",
+    "from varinfo.umm_var import (get_all_umm_var, get_umm_var, publish_all_umm_var,\n",
+    "                             publish_umm_var)\n",
+    "# Singular function call\n",
+    "from varinfo.generate_umm_var import generate_collection_umm_var"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Update authorization header to include your bearer token (e.g. 'Bearer TOKEN')\n",
```

Comment:
Tokens and `Authorization` headers are probably the trickiest part of this whole notebook. I think it would be really helpful to have a markdown cell describing this in a bit of detail.

Reply:
* [ ] Write a markdown cell for elaborating on tokens (edl vs bearer token stuff)

Reply:
Nice - thanks for working on this. It's a tricky topic, and I think the current version helps out a lot.